lib/Doctrine/ORM/Query/SqlWalker.php

```diff
@@ -172,7 +173,9 @@ class SqlWalker implements TreeWalker
     private $quoteStrategy;
 
     /**
-     * {@inheritDoc}
+     * @param Query        $query        The parsed Query.
+     * @param ParserResult $parserResult The result of the parsing process.
+     * @psalm-param array<string, QueryComponent> $queryComponents The query components (symbol table).
```

Comment:
Psalm seems to ignore `@inheritDoc` on constructors. 🤷🏻‍♂️

Reply:
It is supposed to work though: https://github.com/vimeo/psalm/issues/4504#issuecomment-899725382

Reply:
But I think you'd need to add `psalm-consistent-constructor` on the interface as well.

Reply:
Doesn't really work. If I add this annotation to the interface, `@inheritdoc` on `SqlWalker`'s constructor is still ignored. If I remove the attotation from `SqlWalker`, `@inheritdoc` works again, but this also brings back the `UnsafeInstantiation` error for `new $outputWalkerClass`. 😕

Reply:
Ok, I guess defining a constructor in an interface is quite the corner case.